src/utilities/MassIntegrator.cpp

```diff
@@ -0,0 +1,274 @@
+// COPYRIGHT at the top level directory of IBAMR.
+//
+// ---------------------------------------------------------------------
+
+/////////////////////////////// INCLUDES /////////////////////////////////////
+#include "ibamr/MassIntegrator.h"
+#include "ibamr/StaggeredStokesPhysicalBoundaryHelper.h"
+#include "ibamr/ibamr_utilities.h"
+
+#include "ibtk/CartGridFunction.h"
+#include "ibtk/HierarchyGhostCellInterpolation.h"
+#include "ibtk/HierarchyMathOps.h"
+
+#include "BasePatchHierarchy.h"
+#include "BoundaryBox.h"
+#include "Box.h"
+#include "CartesianPatchGeometry.h"
+#include "CoarseFineBoundary.h"
+#include "FaceData.h"
+#include "HierarchyDataOpsManager.h"
+#include "HierarchySideDataOpsReal.h"
+#include "Index.h"
+#include "IntVector.h"
+#include "MultiblockDataTranslator.h"
+#include "Patch.h"
+#include "PatchHierarchy.h"
+#include "PatchLevel.h"
+#include "SideData.h"
+#include "SideGeometry.h"
+#include "SideIndex.h"
+#include "SideVariable.h"
+#include "Variable.h"
+#include "VariableContext.h"
+#include "VariableDatabase.h"
+#include "tbox/Array.h"
+#include "tbox/Database.h"
+#include "tbox/PIO.h"
+#include "tbox/Pointer.h"
+#include "tbox/Timer.h"
+#include "tbox/TimerManager.h"
+#include "tbox/Utilities.h"
+
+#include <array>
+#include <limits>
+#include <memory>
+#include <ostream>
+#include <string>
+#include <utility>
+#include <vector>
+
+#include "ibamr/namespaces.h" // IWYU pragma: keep
+
+namespace SAMRAI
+{
+namespace solv
+{
+template <int DIM>
+class RobinBcCoefStrategy;
+} // namespace solv
+} // namespace SAMRAI
+
+/////////////////////////////// NAMESPACE ////////////////////////////////////
+
+namespace IBAMR
+{
+/////////////////////////////// STATIC ///////////////////////////////////////
+
+namespace
+{
+// NOTE: The number of ghost cells required by the convection scheme depends
+// on the chosen convective limiter, which will be set via input file
+static const int GPPMG = 4;
+static const int GCUIG = 3;
+
+// Timers.
+static Timer* t_apply_convective_operator;
+static Timer* t_integrate;
+static Timer* t_initialize_integrator;
+static Timer* t_deallocate_integrator;
+} // namespace
+
+/////////////////////////////// PUBLIC ///////////////////////////////////////
+MassIntegrator::MassIntegrator(std::string object_name, Pointer<Database> input_db)
+    : d_object_name(std::move(object_name)), d_u_bc_coefs(NDIM), d_rho_bc_coefs(NDIM)
+{
+    if (input_db)
+    {
+        if (input_db->keyExists("bdry_extrap_type"))
+        {
+            d_density_bdry_extrap_type = input_db->getString("bdry_extrap_type");
+        }
+        if (input_db->keyExists("density_bdry_extrap_type"))
+        {
+            d_density_bdry_extrap_type = input_db->getString("density_bdry_extrap_type");
+        }
+        if (input_db->keyExists("convective_limiter"))
+        {
+            d_density_convective_limiter =
+                IBAMR::string_to_enum<LimiterType>(input_db->getString("convective_limiter"));
```

Comment:
Do you need to include the `IBAMR` namespace here?
```suggestion
                string_to_enum<LimiterType>(input_db->getString("convective_limiter"));
```